src/Database/Database.php

```diff
@@ -310,6 +310,57 @@ public function databaseName(): string
 		return $data[0]['db'];
 	}
 
+	/**
+	 * Full-text match on a haystack string that isn't present in the database
+	 *
+	 * @param string $haystack
+	 * @param string $needle Boolean mode search string
+	 * @return bool
+	 * @throws \Exception
+	 */
+	public function performThrowawayFullTextSearch(string $haystack, string $needle): bool
+	{
+		return $this->prepareThrowawayFullTextSearch($haystack, function(string $identifier) use ($needle) {
+			return $this->matchThrowawayFullTextSearch($identifier, $needle);
+		});
+	}
+
+	/**
+	 * Prepare a throwaway full-text search by inserting the haystack string in the temporary table.
+	 * The record is then deleted after the search is expectedly done in the callback function.
+	 * The hopefully unique temporary table identifier is supplied to the callback function.
+	 *
+	 * @param string $haystack Target text to match against
+	 * @param \Closure $callback Signature: function(string $identifier) : mixed
+	 * @return mixed
+	 * @throws \Exception
+	 */
+	public function prepareThrowawayFullTextSearch(string $haystack, \Closure $callback)
```

Comment:
Wouldn't it be better to make a class out of it? Then the entry could be created in the constructor and be removed in the deconstructor.

Reply:
It's as valid, especially since we are using a dedicated table for this. The implementation details would look very similar.

Reply:
At least for me, it would be easier to use (and understand).

Reply:
I can look into that.